src/common/storage/src/config.rs

```diff
@@ -474,3 +478,60 @@ impl ShareTableConfig {
         }
     }
 }
+
+#[derive(Clone, Default, PartialEq, Eq, Debug, Serialize, Deserialize)]
+#[serde(default)]
+pub struct CatalogConfig {
+    #[serde(flatten)]
+    pub catalogs: HashMap<String, CatalogDescription>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "type")]
+pub enum CatalogDescription {
+    #[serde(rename = "hive")]
+    Hive(HiveCatalogConfig),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "protocol")]
+pub enum ThriftProtocol {
+    #[serde(rename = "binary")]
+    Binary,
+    // Compact,
+}
+
+impl FromStr for ThriftProtocol {
```

Comment:
Not needed. `ThriftProtocol` in `common-storage` should always construct by hand or `serde`.